mocks/mocks.go

```diff
@@ -0,0 +1,250 @@
+package mocks
```

Comment:
Wdyt about using https://vektra.github.io/mockery/latest/ (or somthing similar) for generating mock implementation instead of writing them manually?

Reply:
Yeah. @nivida already suggested that library. I am down for it. Will try to incorporate it in this PR.